File: 0-9999/1000-1999/1100-1199/1195/1195.py
import sys 
inpupt = sys.stdin.readline

def is_match(a: str, b: str) -> bool:
    """
    Only a=b=2 does not matched. 
    """
    return not (a == '2' and b == '2')

def solution(A: str, B: str) -> int:
    n, m = len(A), len(B)
    result = n + m

    # gear2를 기준으로 left -> right 이동함 
    for offset in range(-m, n+1):
        ok = True 
        for i in range(m):  
            j = i + offset 
            if 0 <= j < m:
                if not is_match(a=A[j],b=B[i]):
                    ok = False 
        
        if ok:
            left = min(left, offset)
            right = max(right, offset + m)
            result = max(result, right-left)
    
    return result
            

if __name__ == '__main__':
    import sys
    input = sys.stdin.readline
    A = input().strip()
    B = input().strip()
    print(solution(A, B))

File: 0-9999/1000-1999/1100-1199/1195/go/1195.go
package main

import "fmt"

func isMatch(a byte, b byte) bool {
	if a == '2' && b == '2' {
		return false
	}
	return true
}

func solution(A string, B string) int {
	a := len(A)
	b := len(B)
	result := a + b

	for offset := -b; offset <= a; offset++ {
		ok := true

		for i := 0; i < b; i++ {
			j := offset + i
			if 0 <= j && j < a {
				if !isMatch(A[j], B[i]) {
					ok = false
					break
				}
			}
		}

		if ok {
			left := min(0, offset)
			right := max(a, offset+b)
			result = min(result, right-left)
		}
	}

	return result
}

func main() {
	var A string
	fmt.Scanln(&A)

	var B string
	fmt.Scanln(&B)

	fmt.Println(solution(A, B))
}
